### PR TITLE
Interface class of popup message

### PR DESCRIPTION
**Pull request by Agrifon**

### CODE REVIEW

File: Client/Client.Qt/Widgets/I_PopupMessage/PopupMessage.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
+#include "PopupMessage.hpp"
+#include <QPainter>
+#include <QApplication>
+#include <QDesktopWidget>
+#include <QDebug>
+
+PopupMessage::PopupMessage(QWidget* parent) 
+    : I_PopupMessage(parent)
```
Comment (f1nal3) [inline Client/Client.Qt/Widgets/I_PopupMessage/PopupMessage.cpp]: Pls check naming

File: Client/Client.Qt/CMakeLists.txt [hunk 2]
```diff
@@ -80,7 +80,10 @@ set(UI_WIDGETS_SOURCES
         Widgets/TitleWidget.cpp
         Widgets/TitleWidget.hpp
         Widgets/PopupSystemMessage.cpp
-        Widgets/PopupSystemMessage.hpp)
+        Widgets/PopupSystemMessage.hpp
```
Comment (Orange-hanter) [inline Client/Client.Qt/CMakeLists.txt]: Should be removed

File: Client/Client.Qt/CMakeLists.txt [hunk 3]
```diff
@@ -80,7 +80,10 @@ set(UI_WIDGETS_SOURCES
...
+        Widgets/IPopupMessage/IPopupMessage.hpp
+        Widgets/IPopupMessage/PopupMessage.hpp
```
Comment (Orange-hanter) [inline Client/Client.Qt/CMakeLists.txt]: Move to general folder

File: Client/Client.Qt/Pages/RegistrationPage.hpp [hunk 4]
```diff
@@ -4,7 +4,8 @@
 #include <QLabel>
 #include <Utility/Utility.hpp>
 #include <Widgets/Page.hpp>
-#include <Widgets/PopupSystemMessage.hpp>
+#include <Widgets/IPopupMessage/IPopupMessage.hpp>
```
Comment (Orange-hanter) [inline Client/Client.Qt/Pages/RegistrationPage.hpp]: remove

File: Client/Client.Qt/Widgets/IPopupMessage/PopupMessage.hpp [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+#include <QTimer>
+
+class PopupMessage : public IPopupMessage
+{
+public:
+    explicit PopupMessage(QWidget* parent = nullptr);
+
+    virtual ~PopupMessage() override
+    {
+
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/IPopupMessage/PopupMessage.hpp]: Violation of styleguide

File: Client/Client.Qt/Widgets/IPopupMessage/PopupMessage.hpp [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    void hide();
+
+private:
+    QLabel             label;
+    QGridLayout        layout;
+    QPropertyAnimation animation;
+    QTimer*            timer;
+
+    float getPopupOpacity() const override;
+    void  setPopupOpacity(float opacity) override;
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/IPopupMessage/PopupMessage.hpp]: styleguide is dead here!

**Review by Orange-hanter — CHANGES_REQUESTED** [review]

**Pull request closed without merging**